### ISSUE TITLE
torch is not compiled with CUDA support

### ISSUE THREAD

**Issue by wxthu** [pytorch/pytorch#84162]
### 🐛 Describe the bug

when I compiled pytorch source code according to official docs, I found that I cannot run torch inference on GPU. How can I manually control the compiling flags

### Versions

The latest torch version:1.13

cc @malfet @seemethere @ngimel

**Comment by wxthu** [comment]
![image](https://user-images.githubusercontent.com/73052387/187014463-8c05a09a-e7aa-4315-8d04-9b5b34e3f80a.png)

**Comment by malfet** [comment]
Please run `torch.utils.collect_env.py` and share its output or share the configuration log from setup.py

The configuration script is designed to automatically build with CUDA unless one passes `USE_CUDA=OFF` via environment variable. I.e. the fact that it finished without it means that you do not have cuda installed.


Reason for `triage-review` label: I want to advocate for `module: compilation issues` label (and perhaps for revival of `insufficient information` )

**Comment by wxthu** [comment]
> Please run `torch.utils.collect_env.py` and share its output or share the configuration log from setup.py
> 
> The configuration script is designed to automatically build with CUDA unless one passes `USE_CUDA=OFF` via environment variable. I.e. the fact that it finished without it means that you do not have cuda installed.
> 
> Reason for `triage-review` label: I want to advocate for `module: compilation issues` label (and perhaps for revival of `insufficient information` )

output by running _torch.utils.collect_env.py_:
Collecting environment information...
PyTorch version: 1.13.0a0+git15b560a
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Linux Mint 20.2 (x86_64)
GCC version: (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Clang version: Could not collect
CMake version: version 3.22.1
Libc version: glibc-2.31

Python version: 3.9.12 (main, Jun  1 2022, 11:38:51)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.4.0-74-generic-x86_64-with-glibc2.31
Is CUDA available: False
CUDA runtime version: 11.2.152
GPU models and configuration: GPU 0: GeForce GTX 1660 Ti
Nvidia driver version: 460.32.03
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.23.1
[pip3] torch==1.13.0a0+git15b560a
[pip3] torchvision==0.13.1
[conda] blas                      1.0                         mkl  
[conda] cudatoolkit               11.3.1               h2bc3f7f_2    anaconda
[conda] ffmpeg                    4.3                  hf484d3e_0    pytorch
[conda] magma-cuda113             2.5.2                         1    pytorch
[conda] mkl                       2021.4.0           h06a4308_640  
[conda] mkl-include               2022.0.1           h06a4308_117  
[conda] mkl-service               2.4.0            py39h7f8727e_0  
[conda] mkl_fft                   1.3.1            py39hd3c417c_0  
[conda] mkl_random                1.2.2            py39h51133e4_0  
[conda] numpy                     1.23.1           py39h6c91a56_0  
[conda] numpy-base                1.23.1           py39ha15fc14_0  
[conda] pytorch                   1.12.1          py3.9_cuda11.3_cudnn8.3.2_0    pytorch
[conda] pytorch-mutex             1.0                        cuda    pytorch
[conda] torch                     1.13.0a0+git15b560a          pypi_0    pypi
[conda] torchvision               0.13.1               py39_cu113    pytorch

**Comment by malfet** [comment]
@wxthu please re-run the build (in the clean folder) and share the output of the build process ? 
And can you build Hello World CUDA app on your machine?

**Comment by boi-doingthings** [comment]
I am having a similar issue while trying to install Pytorch via below:

`pip3 install torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/cu116`

My fresh Ubuntu install has CUDA 11.7 by default but seems like Pytorch haven't released support for it. Could you suggest a to downgrade my CUDA. I am quite a beginner in setting this all up from scratch.

**Comment by wxthu** [comment]
> @wxthu please re-run the build (in the clean folder) and share the output of the build process ? And can you build Hello World CUDA app on your machine?
[Torch_build.txt](https://github.com/pytorch/pytorch/files/9448870/Torch_build.txt)

build process can be found in this file

**Comment by malfet** [comment]
> > [Torch_build.txt](https://github.com/pytorch/pytorch/files/9448870/Torch_build.txt)
> 
> build process can be found in this file

Important line from this file:
```
-- Could NOT find CUDA (missing: CUDA_INCLUDE_DIRS) (found version "11.7")
```

I.e. looks like you have not installed CUDA correctly.

**Comment by malfet** [comment]
> I am having a similar issue while trying to install Pytorch via below:
> 
> `pip3 install torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/cu116`
> 
> My fresh Ubuntu install has CUDA 11.7 by default but seems like Pytorch haven't released support for it. Could you suggest a safe method to downgrade my CUDA. I am quite a beginner in setting this all up from scratch.

This issue is completely different: author of this one is trying to build PyTorch from source, while you are are using a released binaries (which should be forward compatible with CUDA-11.7 driver).

**Comment by malfet** [comment]
Based on the information available so far, it does not looks like a bug in PyTorch build system, but rather an issues with the local setup. Please use https://discuss.pytorch.org/ to ask general questions about PyTorch build system or seek help with the configuration. 

Please do not hesitate to reopen the issue(or file a new one) if you believe that build failures are due to the bug in PyTorch

**Comment by wxthu** [comment]
> Based on the information available so far, it does not looks like a bug in PyTorch build system, but rather an issues with the local setup. Please use https://discuss.pytorch.org/ to ask general questions about PyTorch build system or seek help with the configuration.
> 
> Please do not hesitate to reopen the issue(or file a new one) if you believe that build failures are due to the bug in PyTorch

Thanks for your answer. I have solved the problem by removing **_cuda-nvcc_** package in my condo environment
